cli/daemon/dash/dashapp/src/components/api/RPCCaller.tsx

```diff
@@ -602,28 +499,36 @@ const RPCPathEditor = React.forwardRef<{getPath: () => string | undefined}, {
 
 // encodeQuery encodes a payload matching the given schema as a query string.
 // If the payload can't be parsed as JSON it throws an exception.
-function encodeQuery(schema: Decl, payload: string): string {
+function encodeQuery(md: APIMeta, schema: Type, payload: string): string {
   const json = JSON.parse(payload)
   let pairs: string[] = []
 
-  for (const f of schema.type.struct?.fields ?? []) {
-    let key = f.json_name
-    let qsName = f.query_string_name
-    if (key === "-" || qsName === "-") {
-      continue
-    } else if (key === "") {
-      key = f.name
-    }
+  if (schema.named) {
+    const declID = schema.named.id
+    const decl = md.decls[declID]
+
+    for (const f of decl.type.struct?.fields ?? []) {
+      let key = f.json_name
+      let qsName = f.query_string_name
+      if (key === "-" || qsName === "-") {
+        continue
+      } else if (key === "") {
+        key = f.name
+      }
 
-    let val = json[key]
-    if (typeof val === "undefined") {
-      continue
-    } else if (!Array.isArray(val)) {
-      val = [val]
-    }
-    for (const v of val) {
-      pairs.push(`${qsName}=${encodeURIComponent(v)}`)
+      let val = json[key]
+      if (typeof val === "undefined") {
+        continue
+      } else if (!Array.isArray(val)) {
+        val = [val]
+      }
+      for (const v of val) {
+        pairs.push(`${qsName}=${encodeURIComponent(v)}`)
+      }
     }
+  } else {
+    throw new Error('expected a named type to encode the query');
```

Comment:
Not without moving an error state higher in the stack annoyingly. Practically though this shouldn't happen because the Go system won't produce an app with a query string that isn't named.